### PR TITLE
NPM Release: Preparing NPM release

### PR DESCRIPTION
**Pull request by dlanthier-mac**
Created a plugin with plugman to create the Cordova Plugin.
Updated the gitignore.
Renamed the plugin to be compatible with npm package names.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "cordova-blinkup-plugin",
+  "version": "0.5.0",
+  "description": "\n        This plugin allows you to integrate the native BlinkUp process for setting up an Electric Imp device directly into your Cordova / PhoneGap project.\n        \n        The Plan ID, Device ID, Agent URL, Verfication Date, and status of the operation are all passed back as JSON to the Cordova application, along with a status or error code (see ReadMe.md for JSON format and status code definitions).\n    ",
```
Comment (sdouglas-macadamian) [inline package.json]: remove the newline and leading spaces?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    "blinkup"
+  ],
+  "engines": [
+    {
+      "name": "cordova",
+      "version": ">=3.4.0"
+    }
+  ],
+  "author": "",
+  "license": "\n        Licensed under the Apache License, Version 2.0 (the \"License\");\n        you may not use this file except in compliance with the License.\n        You may obtain a copy of the License at\n        \n        http://www.apache.org/licenses/LICENSE-2.0\n        \n        Unless required by applicable law or agreed to in writing, software\n        distributed under the License is distributed on an \"AS IS\" BASIS,\n        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n        See the License for the specific language governing permissions and\n        limitations under the License.\n        \n        Created by Stuart Douglas (sdouglas@macadamian.com) on June 11, 2015.\n        Copyright (c) 2015 Macadamian. All rights reserved.\n    ",
```
Comment (sdouglas-macadamian) [inline package.json]: leading newline and spaces

File: plugin.xml [hunk 3]
```diff
@@ -97,7 +97,7 @@
     <!--========== iOS ==========-->
     <platform name="ios">
         <config-file target="config.xml" parent="/*">
-            <feature name="BlinkUpPlugin">
+            <feature name="cordova-blinkup-plugin">
```
Comment (sdouglas-macadamian) [inline plugin.xml]: do we need to rename BlinkUpPlugin.m / BlinkUpPlugin.h ?

File: plugin.xml [hunk 4]
```diff
@@ -63,7 +63,7 @@
     <!--========== Android ==========-->
     <platform name="android">
         <config-file target="res/xml/config.xml" parent="/*">
-            <feature name="BlinkUpPlugin">
+            <feature name="cordova-blinkup-plugin">
```
Comment (sdouglas-macadamian) [inline plugin.xml]: May need to rename BlinkUpPlugin.java. If so, maybe just rename the <name> (as above), but don't change the <feature>'s

File: plugin.xml (hunk 3 above)
Comment (dlanthier-mac) [inline plugin.xml]: no

File: package.json (hunk 2 above)
Comment (dlanthier-mac) [inline package.json]: This was autogenerated by plugman.

File: plugin.xml (hunk 4 above)
Comment (dlanthier-mac) [inline plugin.xml]: No, these are different things. The value drives what the entry class name is.

**Comment by dlanthier-mac** [comment]
addressed comments.

**Pull request merged**